pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -28,6 +28,44 @@
 INV_SQRT2 = 1 / qml.math.sqrt(2)
 
 
+class Barrier(Operation):
+    r"""Barrier(wires)
+    The Barrier operator, separates the compilation process into blocks or is used as a visual tool.
+
+    **Details:**
+
+    * Number of wires: AnyWires
+    * Number of parameters: 0
+
+    Args:
+        only_visual (bool): True if we do not want it to have an impact on the compilation process. Default is False.
+        wires (Sequence[int] or int): the wires the operation acts on
+    """
+
+    def __init__(self, only_visual=False, *params, wires=None, do_queue=True, id=None):
+
+        self.only_visual = only_visual
+        super().__init__(*params, wires=wires, do_queue=do_queue, id=id)
+
+    num_params = 0
+    num_wires = AnyWires
+    par_domain = None
+
+    def decomposition(self, wires):
+        _wires = wires
+        if self.only_visual:
+            decomp_ops = []
+        else:
+            raise NotImplementedError
+        return decomp_ops
+
+    def _matrix(self, *params):
+        return np.identity(2 ** len(self.wires))
```

Comment:
Not sure if we'd like a matrix to be returned here. :thinking: Most underlying devices will want to use this matrix to apply the operation. For a noisy device, applying an identity might lead to noise and from an abstraction point of view we're applying an identity for an operation that "doesn't do anything".

What if we returned an empty list/empty array here?